packages/hooks/use-escape-keydown/index.ts

```diff
@@ -1,19 +1,32 @@
 import { onBeforeUnmount, onMounted } from 'vue'
-import { off, on } from '@element-plus/utils'
+import { isClient } from '@vueuse/core'
 import { EVENT_CODE } from '@element-plus/constants'
 
-export const useEscapeKeydown = (handler?: (e: KeyboardEvent) => void) => {
+let registeredEscapeHandlers: ((e: KeyboardEvent) => void)[] = []
+
+export const useEscapeKeydown = (handler: (e: KeyboardEvent) => void) => {
   const cachedHandler = (e: Event) => {
     const event = e as KeyboardEvent
     if (event.key === EVENT_CODE.esc) {
-      handler?.(event)
+      registeredEscapeHandlers.forEach((registeredHandler) =>
+        registeredHandler(event)
+      )
     }
   }
+
   onMounted(() => {
-    on(document, 'keydown', cachedHandler)
+    if (registeredEscapeHandlers.length === 0) {
+      document.addEventListener('keydown', cachedHandler)
+    }
+    if (isClient) registeredEscapeHandlers.push(handler)
   })
```

Comment:
IMHO, `onMounted` will only get invoked on client side, so I do not think it is necessary to check that here.

Reply:
Yeah I just moved that isClient check since it was there in the previous code